src/iconv.erl

```diff
@@ -3,27 +3,39 @@
 % Licensed under the Apache License, Version 2.0 (the "License");
 % you may not use this file except in compliance with the License.
 % You may obtain a copy of the License at
-% 
+%
 %     http://www.apache.org/licenses/LICENSE-2.0
-% 
+%
 % Unless required by applicable law or agreed to in writing, software
 % distributed under the License is distributed on an "AS IS" BASIS,
 % WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 % See the License for the specific language governing permissions and
 % limitations under the License.
 
-% drop in replacement module. 
+% drop in replacement module.
 
 -module(iconv).
 -author("Maas-Maarten Zeeman <mmzeeman@xs4all.nl>").
 
--export([open/2, conv/2, close/1]).
+-export([open/2, conv/2, close/1, convert/3]).
 
-open(ToCode, FromCode) -> 
+open(ToCode, FromCode) ->
     eiconv:open(ToCode, FromCode).
 
-conv(Cd, Input) -> 
+conv(Cd, Input) ->
     eiconv:conv(Cd, Input).
 
-close(Cd) -> 
+close(Cd) ->
     eiconv:close(Cd).
+
+% @doc Convert input which is in FromEncoding to ToEncoding.
+%
+convert(FromEncoding, ToEncoding, Input) ->
+    case eiconv:convert(FromEncoding, ToEncoding, Input) of
+        {ok, Output} -> Output;
+        {error, Error} ->
+            % throw the original error.
+            % TODO Let the badmatch occur?
+            % But it is not easy to catch by the user.
+            erlang:throw(Error)
```

Comment:
Yes, this is for faulty inputs. As it is used for text encoding and may be submitted text the user do not control (like emails parsed by gen_smtp), it should be easy to handle the error case.

Actually, this is the interface adopted by `eiconv:convert/3` : https://github.com/zotonic/eiconv/blob/cf2c4b1d2e18da64cc691867b75143697b88980e/src/eiconv.erl#L103

I also find this interface more appropriate.

However here I added this `iconv/convert:3` function as a wrapper to be compatible with other iconv libraries. These other libraries ([processone/iconv](https://github.com/processone/iconv), [emqplus/erlang-iconv/](https://github.com/emqplus/erlang-iconv/blob/master/src/iconv.erl#L92)) have chosen as interface to return directly the output text.
(See Vagabond/gen_smtp/pull/132 for more context.)

I'm not sure we can return an `{error, ...}` for the error case and return the decoded text directly for the nominal case.

Note that the current `eiconv:convert/3` interface remains unchanged. Addind a wrapper here seems easier than submitted changes in other libraries that will break their APIs.